src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,89 @@
 package core.basesyntax;
 
+import java.util.Objects;
+
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int DEFAULT_CAPACITY = 16;
+    private static final int INCREASE_FACTOR = 2;
+    private static final float LOAD_FACTOR = 0.75f;
+    private int size;
+    private final int threshold;
+    private Node<K, V>[] table;
+
+    public MyHashMap() {
+        table = new Node[DEFAULT_CAPACITY];
+        threshold = (int) (DEFAULT_CAPACITY * LOAD_FACTOR);
+    }
 
     @Override
     public void put(K key, V value) {
-
+        if (size == threshold) {
+            enlargeTable();
+        }
+        int index = getNodeIndex(key);
+        Node<K, V> node = table[index];
+        if (node == null) {
+            table[index] = new Node<>(key, value);
+            size++;
+        }
+        while (node != null) {
+            if (Objects.equals(key, node.key)) {
+                node.value = value;
+                return;
+            }
+            if (node.next == null) {
+                node.next = new Node<>(key, value);
+                size++;
+                return;
+            }
+            node = node.next;
+        }
     }
 
     @Override
     public V getValue(K key) {
+        Node<K, V> element = table[getNodeIndex(key)];
+        while (element != null) {
+            if (Objects.equals(key, element.key)) {
+                return element.value;
+            }
+            element = element.next;
+        }
         return null;
     }
 
     @Override
     public int getSize() {
-        return 0;
+        return size;
+    }
+
+    private void enlargeTable() {
+        Node<K, V>[] oldTable = table;
+        table = new Node[DEFAULT_CAPACITY * INCREASE_FACTOR];
```

Comment:
```suggestion
        table = new Node[oldTable.length * INCREASE_FACTOR];
        treshold *= INCREASE_FACTOR;
```
this method would always create table with size 32
treshold also should be changed